src/v/pandaproxy/test/kafka_client_cache.cc

```diff
@@ -125,6 +127,24 @@ SEASTAR_THREAD_TEST_CASE(cache_fetch_or_insert) {
     BOOST_TEST(client2->config().scram_password.value() == user2.pass);
     BOOST_TEST(client_cache.size() == s);
     BOOST_TEST(client_cache.max_size() == max_s);
+}
 
-    client_cache.g.close().get();
+SEASTAR_THREAD_TEST_CASE(test_keep_alive) {
+    pp::credential_t user{"red", "panda"};
+    size_t cache_max_size = 5;
+    std::chrono::milliseconds keep_alive{50ms};
+
+    pp::test_client_cache client_cache{cache_max_size, keep_alive};
+    client_cache.start().get();
+
+    // This will insert a client into the cache. The client is keyed by the user
+    // name.
+    client_cache.fetch_or_insert(user, config::rest_authn_method::none);
```

Comment:
Isn't there technically a race here between insert and cleanup code? If the cleanup code stops half way through, and then an insert, and then cleanup code continues, could any assumptions about the iterators be invalided during the cleanup?

Reply:
I see. Because it's sync, and the way that cleanup is formulated to be safe across suspension points, this is ok.

Reply:
Yeah it should be safe but I can also be more safe by changing the order:
- Call `fetch_of_insert` first
- Then call `cache.start` because all the start method does is start the GC timer.

Reply:
That's probably not particularly representative of how it will be used in Redpanda, though. It's either safe or it's not, and I believe it is.